drivers/flash/soc_flash_nrf.h

```diff
@@ -0,0 +1,160 @@
+				     * synchronization backend. It is used to
+				     * transfer timing parameters and store
+				     * data required by the backend.
+				     * Structure of contex depends on backend
+				     * implementation.
+				     */
+#endif /* !CONFIG_SOC_FLASH_NRF_RADIO_SYNC_NONE */
+#if defined(CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE)
+	uint32_t flash_addr_next;
+#endif /* CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE */
+}; /*< Context type for f. @ref write_op @ref erase_op */
+
+#ifndef CONFIG_SOC_FLASH_NRF_RADIO_SYNC_NONE
+/**
+ * @defgroup nrf_flash_synch synch backend API
+ *
+ * API declared bellow contains prototypes of function which shall be
+ * implemented by the synchronization backend.
+ * @{
+ */
+
+/**
+ * Calback which executes the flash operation.
+ *
+ * @param context pointer to flash_context structure.
+ * @retval @ref FLASH_OP_DONE once operation was done, @ref FLASH_OP_ONGOING if
+ *         opperation needs more time for execution.
+ */
+typedef int (*flash_op_handler_t) (void *context);
+
+struct flash_op_desc {
+	flash_op_handler_t handler;
+	struct flash_context *context; /* [in,out] */
+};
+
+/**
+ * Synchronization backend driver initialization procedure.
+ *
+ * This will be run within flash driver initialization
+ */
+int nrfsb_synch_backend_init(void);
+
+/**
+ * Give synchronization context for write operation.
+ *
+ * @retval Synchronization context for the write operation.
+ * Structure of contex depends on backend implementation.
+ */
+void *nrfsb_get_write_synch_context(void);
+
+/**
+ * Give synchronization context for erase operation.
+ *
+ * @retval Synchronization context for the erase operation.
+ */
+void *nrfsb_get_erase_synch_context(void);
+
+/**
+ * Check if the operation need to be run synchonous with radio.
+ *
+ * @retval True if operation need to be run synchronously, otherwise False
+ */
+bool nrfsb_is_synchronization_required(void);
+
+/**
+ * Execute the flash opperation synchronously along the radio operations.
+ *
+ * Function executes calbacks op_desc->handler() in execution windows according
+ * to timing settings transfereed by op_desc->context->synch_context.
+ * This routine need to be called the handler as many time as it returns
+ * FLASH_OP_ONGOING, howewer an operation timeot should be implemented.
+ * When the handler() returns FLASH_OP_DONE, no further execution windows are
+ * needed so function should return as the handler() finished its operation.
+ *
+ * @retval 0 if op_desc->handler() was executed and
+ * finnished its opperation. Otherwise (timeout, couldn't schedule execution...)
+ * a negative errorcode.
+ */
+int nrfsb_work_synchronous_with_radio(struct flash_op_desc *op_desc);
+
+/**
+ * @}
+ */
+
+/**
+ * @defgroup nrf_flash_synch_timing synch timing backend API
+ * @ingroup nrf_flash_synch
+ * @{
+ *
+ * API which is used by nrf flash driver for check where execution fill in
+ * the execution window.
+ *
+ * API is used as fallow:
+ * begin of execution window
```

Comment:
Could you add to this example also calls to:
* `nrfsb_get_write_synch_context`
* `nrfsb_work_synchronous_with_radio`
* `flash_op_handler_t handler`
and mention which function is called by which module?

I think it would clarify how this API is used and how to implement it.

Reply:
>Could you add to this example also calls to:
* `nrfsb_get_write_synch_context`
* `nrfsb_work_synchronous_with_radio`

Will improve that in the next update